Support custom value for marketplace in integrations and script

ilaner opened the pull request:
This supports setting values for keys for specific marketplace.
For instance, we will be able to set `defaultValue` to `true`, but override this value in XPANSE with `defaultValue:xpanse: false`.

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-8959

## Description

yuvalbenshalom commented on demisto_sdk/commands/common/schemas/classifier.yml:
```diff
@@ -18,44 +18,85 @@ mapping:
+    type: bool
+  regex;(definitionId:(xsoar)|(marketplacev2)|(xpanse)|(xsoar_on_prem)|(xsoar_saas)):
+    type: str
+  regex;(marketplaces:(xsoar)|(marketplacev2)|(xpanse)|(xsoar_on_prem)|(xsoar_saas)):
```
is there a point to add this to marketplaces?

yuvalbenshalom commented on demisto_sdk/commands/content_graph/objects/content_item.py:
```diff
@@ -221,9 +221,7 @@ def prepare_for_upload(
         data = self.data
         logger.debug(f"preparing {self.path}")
-        return MarketplaceSuffixPreparer.prepare(
-            data, current_marketplace, self.marketplaces
```
how is this change relevant?

ilaner replied:
This argument wasn't used and is not relevant.
We prepare the content item for a specific marketplace, and we need to provide only the marketplace we work on, and not all the marketplaces this content item supports.

yuvalbenshalom commented on demisto_sdk/commands/prepare_content/preparers/marketplace_suffix_preparer.py:
```diff
@@ -26,37 +23,51 @@ def prepare(
+                        continue
+                    for suffix in suffixes:
+                        # iterate each suffix to see if it's relevant for the key.
+                        # the order of the suffixes matter!
```
lets go over the cases, I like to understand the order importance.

ilaner replied:
OK.
In general: if running on XSOAR_SAAS and we have the following keys:
<key>:xsoar_saas
<key>:xsoar
<key>

We should priorotize the xsoar_saas key over the xsoar. However, if there is only xsoar, we should take it and not the regular key.

yuvalbenshalom commented on demisto_sdk/commands/prepare_content/tests/marketplace_suffix_preparer_test.py:
```diff
@@ -0,0 +1,119 @@
+}
+
+
+def test_remove_xsoar():
```
please add docstrings explaining the cases.

yuvalbenshalom commented on demisto_sdk/scripts/add_marketplace_to_schemas.py:
```diff
@@ -0,0 +1,52 @@
+from copy import deepcopy
+from pathlib import Path
```
let's talk about what ever this should be in the repo, and if so, it shall include UTs.

yuvalbenshalom reviewed (commented):
looks ok, but I like us to discuss if we are really interested in adding all posibblties to all schemas, as it is complex and 99% of it will probably never be used.

ilaner commented:
Test plan:
1. Added unit-tests for the suffix preparer, to run on custom data (not content-like)
2. Fixed unit-tests for content data to handle the new syntax and not _x2.
3. run prepare-content with all markteplaces, and show that the value in the key was modified in the result.
4. Existing schemas still work: validate -a passes
5. Added tests for the new schema syntax, make sure that <supported_key>:<marketplace> passes validation
